runner/output.go

```diff
@@ -0,0 +1,24 @@
+package runner
+
+import (
+	"io"
+)
+
+// Saver lets clients create new Outputs so they can save data.
+// This is how Runner can save stdout and stderr.
+// (NB: Saver is temporary until we save output into a new Snapshot)
+type Saver interface {
```

Comment:
Why not have these two interfaces be combined.  This seems like you just want a Output to have New or Create method associated with it.  

Or is this separate because of how you envision the snapshot migration?